scriptworker/task.py

```diff
@@ -462,10 +455,33 @@ def is_try_or_pull_request(context, task):
     Returns:
         bool: True if it's a pull-request or a try task
 
+    """
+    if is_github_task(task):
+        return await is_pull_request(context, task)
+    else:
+        return is_try(task, context.config['source_env_prefix'])
+
+
+def is_github_task(task):
+    """Determine if a task is related to GitHub.
+
+    This function currently looks into the ``schedulerId``, ``extra.tasks_for``, and
+    ``metadata.source``.
+
+    Args:
+        task (dict): the task definition to check.
+
+    Returns:
+        bool: True if a piece of data refers to GitHub
+
     """
     return any((
-        is_try(task, context.config['source_env_prefix']),
-        is_pull_request(context, task),
+        # XXX Cron tasks don't usually define 'taskcluster-github' as their schedulerId as they
```

Comment:
Once we use standalone taskgraph in our github repos, we can look at standardizing how the taskgraph generates cron tasks so we can detect it here.